### ISSUE TITLE
[export] incorrect report on `export * as`

### ISSUE THREAD

**Issue by JounQin** [benmosher/eslint-plugin-import#2002]
```js
// processors/remark.js
export const remark = {}

// processors/index.js
export * as processors from './remark'

// rules/remark.js
export const remark = {}

// rules/index.js
import { remark } from './remark'

export { remark }
export const rules = { remark }

// index.js
export * from './processors'
export * from './rules'
```

Unexpected error:

```log
Multiple exports of name 'remark'
```

But `processors` doesn't export `remark` at all.

---

Workaround:

```js
// processors/index.js
import { remark } from './remark'

export const processors = { remark }
```

**Comment by jsg2021** [comment]
From your snippet we can clearly see an export const remark, and a little below an `export { remark }`... that's what the error is catching.

**Comment by ljharb** [comment]
Indeed, this looks like a correct error. Thanks @jsg2021 for identifying that!

**Comment by jsg2021** [comment]
@ljharb sorry for the noise. i deleted my comment after realizing the snippet is of several files not one.

**Comment by JounQin** [comment]
😂 Please reopen it.

**Comment by jsg2021** [comment]
@JounQin if you could make a sample repo with the problem it would help or a failing test in a PR.

**Comment by JounQin** [comment]
@jsg2021 Here you are: https://github.com/JounQin/test/tree/eslint-plugin-import

Simply run: `yarn lint`

**Comment by devfservant** [comment]
Having the same “Multiple exports” false positive issue with a slightly different situation:  
(where `remark` is not even exported from *rules/index.js*)

```
root
│   index.ts
│
└───bar
│   │   bar.utils.ts
│   │   index.ts
│
└───foo
    │   foo.utils.ts
    │   index.ts
```


```ts
// bar/bar.utils.ts
export const getOptions = (): any => {};
```

```ts
// bar/index.ts
export * as barUtils from './bar.utils';
```

```ts
// foo/foo.utils.ts
export const getOptions = (): any => {};
```

```ts
// foo/index.ts
export * as fooUtils from './foo.utils';
```

```ts
// index.ts
export * from './bar';
export * from './foo';
```

```
/../root/index.ts
  1:1  error  Multiple exports of name 'getOptions'  import/export
  2:1  error  Multiple exports of name 'getOptions'  import/export
✖ 2 problems (2 errors, 0 warnings)
```

**Comment by kristiannotari** [comment]
Having the same issue as @devfservant stated. The conflicting names are behind a named export, hence they should not be conflicting at all
